app/controllers/exercises_controller.rb

```diff
@@ -150,6 +183,44 @@ def implement
     end
   end
 
+  def set_course_token
+    lti_parameters = LtiParameter.find_by(external_users_id: current_user.id,
+                                          exercises_id: @exercise.id)
+    if lti_parameters
+      lti_json = lti_parameters.lti_parameters["lis_outcome_service_url"]
+      @course_token =
+          if match = lti_json.match(/^.*courses\/([a-z0-9\-]+)\/sections/)
```

Comment:
put this into a class "platform adapter" (for example: openhpi_platform_adapter), that knows specifics about consumers, such as URL patterns etc.